.travis.yml

```diff
@@ -1,4 +1,31 @@
+sudo: false
 language: rust
+addons:
+  apt:
+    packages:
+      - libcurl4-openssl-dev
+      - libelf-dev
+      - libdw-dev
+rust:
+  - nightly
+  - beta
+  - stable
+  - 1.3.0
+matrix:
+  allow_failures:
+    - rust: nightly
+before_script:
+  - |
+      pip install 'travis-cargo<0.2' --user &&
+      export PATH=$HOME/.local/bin:$PATH
 script:
-  - cargo test
-sudo: false
+  - |
+      travis-cargo build &&
+      travis-cargo test &&
+      travis-cargo bench &&
+      travis-cargo --only stable doc
+after_success:
+  - travis-cargo --only stable doc-upload
```

Comment:
If possible, since your in here, lets add auto publishing at the end of the build, I will add the token to travis.  I believe this will do it

```yml
  - [ $TRAVIS_BRANCH = master ] && [ $TRAVIS_PULL_REQUEST = false ] && cargo publish --token $CARGO_TOKEN
```